### DISCUSSION TITLE
Pokédex listo!!

### DISCUSSION THREAD

**Discussion by marco-cortes** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#428] [Show and tell]
![image](https://user-images.githubusercontent.com/84252964/158713778-5042adde-8ef2-424a-ba21-b11a19675147.png)

Práctica de Js finalizada :)
Por si quieren probarla: https://marco-cortes.me/pokedex

**Comment by pavg92** [comment]
Te quedó muy bien 👍

**Comment by RamSantoyo** [comment]
me mamo tu interfaz, pero creo que si pones las imágenes que vienen en "data.sprites.other.home.front_shin" se verían mejor, tienen mas resolución

**Reply by sergiolopezloya** [reply]
El shin está en las otras resoluciones más chicas, ya lo probé en el mio y no se ve nada, igual en esa imagen que se muestra se ve muy bien

**Comment by LuisEsquivel27** [comment]
Se ve bien!! podrías explicarme como realizaste el cambio de estilos de acuerdo al pokemon por favor?

**Reply by marco-cortes** [reply]
holaa, lo que hago es que uso variables para los colores tanto de la tarjeta, como del formulario y el fondo y dependiendo el primer tipo de pokemon que traiga la api, cambio el valor de esa variable, esto lo hago con clases que se llaman del mismo nombre que los tipos de pokemon, te dejo mi css para que veas las clases que hay y las variables de colores: https://github.com/marco-cortes/pokedex/blob/main/styles.css

**Reply by LuisEsquivel27** [reply]
gracias, lo voy a revisar

**Comment by Ingrzs** [comment]
La parte de los rango? Como lo rellenaste ?
Como lo pasaste de JavaScript a html? Estoy haciendo eso similar

**Reply by marco-cortes** [reply]
Un div con un background-color clarito y dentro un span con el background-color más fuerte
El div tiene un ancho de 100% y el span dependiendo el valor del dato tendrá el ancho ;b

https://github.com/marco-cortes/pokedex/blob/main/app.js

Lo del js lo hago de la línea 42 a la 47
Los elementos que invoco por las clases son el div y el span jeje

**Reply by Ingrzs** [reply]
Impresionate lo que has hecho, tu conviertes los valores a tostring , algo que aun no cónozco, tengo que estudiar mas 😅

**Comment by GloriaCoralCerecedo** [comment]
Hola, te quedo genial!, ¿como muestras las estadísticas?

**Reply by marco-cortes** [reply]
Tengo dos listas, una para los títulos, los de HP, ATK, etc… y otra para los valores y las barras de progreso, para las barras de progreso hago lo siguiente:

![image](https://user-images.githubusercontent.com/84252964/158834402-03158813-d060-4f15-9873-4284be1d8f41.png)

Tengo un div y un span, el div tiene la clase "stat-bar", esta será la barra completa que tiene un color más clarito, y tendrá el width de 100%, los estilos son los siguientes:

![image](https://user-images.githubusercontent.com/84252964/158834471-d127e303-3654-41fa-916e-522f7cec1194.png)

Después dentro del div, esta el span con la clase "bar" será la barra de progreso y tendrá un color más oscuro, los estilos son los siguientes:

![image](https://user-images.githubusercontent.com/84252964/158834503-d80790d3-1e92-4a19-bda7-40c22e505f8c.png)

Ahora, para que se vea como la barra de progreso, los span con la clase "bar" deben tener un width, el cual se los pongo con el siguiente JS:

![image](https://user-images.githubusercontent.com/84252964/158834547-26f4ad75-3ca5-4208-b06c-43b595dc3447.png)

Este JS lo que hace es que toma el valor de los span con la clase "progress" y los establece como width a los span de la clase "bar", para que así se muestre el "progreso" y yap :)

**Reply by GloriaCoralCerecedo** [reply]
Oh ya sacas la info del API, la verdad entendí masomenos peor gracias jaja

**Comment by EslebanPerez** [comment]
Te quedó genial! 👍👍

**Comment by sergiolopezloya** [comment]
Se ve genial, tomé de referencia unos acomodos que tienes, en general me gustó mucho
